### PR TITLE
Migration of common code from Cinderscapes to Terraform

### PR DESCRIPTION
**Pull request by ghost**
This PR moves over the following items from Cinderscapes to Terraform as we've discussed. Each of these items seems to fit better in Terraform than they do in Cinderscapes.

- Created a separate nether biome builder
- Moved the mixed noise point checker
- Moved the map biomes command*
- Moved the fog density code because there was interest in using it in overworld biomes eventually

* SuperCoder is going to rewrite this after it gets merged because the current version is a little lack-luster, but its included in the PR

### CODE REVIEW

File: src/main/java/com/terraformersmc/terraform/Terraform.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package com.terraformersmc.terraform;
+
+import com.terraformersmc.terraform.command.MapBiomesCommand;
+import com.terraformersmc.terraform.util.NoiseCollisionChecker;
+import net.fabricmc.api.ModInitializer;
+
+import java.util.logging.Logger;
+
+/**
```
Comment (NeusFear) [inline src/main/java/com/terraformersmc/terraform/Terraform.java]: I don't think we need docs here

File: src/main/java/com/terraformersmc/terraform/command/MapBiomesCommand.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+		//setup image
+		Map<Biome, Integer> biomeCount = new HashMap<>();
+		BufferedImage img = new BufferedImage(2048, 2048, BufferedImage.TYPE_INT_RGB);
+
+		int progressUpdate = img.getHeight() / 8;
+
+		for (int x = 0; x < img.getHeight(); x++) {
+			for (int z = 0; z < img.getWidth(); z++) {
+				Biome b = source.getWorld().getBiomeForNoiseGen(x * 4, 0, z * 4);
```
Comment (NeusFear) [inline src/main/java/com/terraformersmc/terraform/command/MapBiomesCommand.java:67]: Don't instantiate variables inside loops

**Review by NeusFear — CHANGES_REQUESTED** [review]
Looks pretty good to me, these two fixes should take like 3 seconds to apply

**Comment by ghost** [comment]
Are we still even looking at doing this PR right now or are we going to modularize Terraform first.

Also the second part about the variable inside the loop was a Scoder thing, we were going to just merge the PR and then have him rewrite the map biomes command after that.

**Comment by NeusFear** [comment]
We may as well merge it, the api for biome building should stay the same I'd think

**Comment by ghost** [comment]
Sounds good one sec I’ll remove the doc with the web editor :P

**Comment by ghost** [comment]
Okay it should be good now

**Pull request merged**